### DISCUSSION TITLE
Wrong MMFF atom types with aromatic rings

### DISCUSSION THREAD

**Discussion by apiserchia** [rdkit/rdkit#5625] [Q&A]
Given a SMILE containing aromatic rings, MMFF methods use their own aromaticity model therefore changing molecule's aromaticity. In this way I get wrong MMFF94 atom types and energy contributions. I read on the following article ["Bringing the MMFF force field to the RDKit: implementation and validation"](https://jcheminf.biomedcentral.com/articles/10.1186/s13321-014-0037-3) that

> "Since the aromaticity model used by MMFF differs from the one normally used throughout the RDKit, aromaticity has to be re-perceived according to MMFF criteria starting from a kekulized representation of the molecule. Subsequently, atom types are assigned to heavy atoms followed by hydrogens."

Also [RDKit Documentation](https://rdkit.readthedocs.io/en/latest/GettingStartedInPython.html) clearly states that

> "Please note that the MMFF atom typing code uses its own aromaticity model, so the aromaticity flags of the molecule will be modified after calling MMFF-related methods."

Is there a method/way to retrieve correct MMFF94 atom types and energy contributions from a given SMILE? If yes, what should I do? Am I missing something in the following snippet of Python code?

**To Reproduce**

```
from rdkit import Chem
from rdkit.Chem import AllChem

#Thymine
molecule = Chem.MolFromSmiles('O=C1NC(=O)NC=C1C')

moleculewH = Chem.AddHs(molecule)

AllChem.EmbedMolecule(moleculewH)

properties = AllChem.MMFFGetMoleculeProperties(moleculewH)

for atom in range(moleculewH.GetNumAtoms()):
    print(atom+1, "\t", properties.GetMMFFAtomType(atom), "\t", moleculewH.GetAtomWithIdx(atom).GetIsAromatic())

properties.SetMMFFVerbosity(2)
forcefield = AllChem.MMFFGetMoleculeForceField(moleculewH, properties)
```

Here below I attach an output screenshot printed from the above snippet of Python code. Only [atom | MMFF94 atom type | is atom aromatic] and bond stretching contributions are shown for sake of brevity.

![Immagine 2022-10-06 172619](https://user-images.githubusercontent.com/83699346/194364533-33b58dff-15a8-4f90-a062-1984936198b0.jpg)

I would expect to see right MMFF94 atom types for aromatic atoms and ring atoms to be recognized as aromatic.
I have the same problem e.g. for all nucleobases and also for thymine expressed in different ways such as `CC1=CNC(=O)NC1=O` or `O=c1[nH]c(=O)[nH]cc1C`

I'm using

    RDKit version: 2022.03.4
    OS: Ubuntu 20.04
    Python version (if relevant): 3.9.1
    Are you using conda? yes
    If you are using conda, which channel did you install the rdkit from? pip
    If you are not using conda: how did you install the RDKit?

Thank you in advance for any help!

**Comment by ptosco** [comment]
Ciao Andrea,

The atom types assigned by RDKit are correct, and so are partial charges; I have just verified the assignment with MOE and OpenBabel I get the same atom types and partial charges:
```
$ obenergy -ff MMFF94 thymine.mol

A T O M   T Y P E S

IDX     TYPE    RING
1       7       NO
2       3       AL
3       10      AL
4       3       AL
5       7       NO
6       10      AL
7       2       AL
8       2       AL
9       1       NO
10      28      NO
11      28      NO
12      5       NO
13      5       NO
14      5       NO
15      5       NO
```
Thymine is not aromatic according to the MMFF aromaticity model, which is why after assigning MMFF atom types the aromaticity flags are all set to `False`.

**Reply by apiserchia** [reply]
Dear Paolo,
thank you for your kind answer. I made a big mistake because I was thinking in "chemical terms", that is, since thymine is aromatic from the chemical point of view, then carbon atoms n.7 and n.8 should be aromatic carbons (MMFFTYPE: 37) instead of olefinic ones (MMFFTYPE: 2), as well as the other ring atoms having more "aromatic" atom types, but I didn't realize that MMFF aromaticity model excluded such molecules from being aromatic.
I understand this is due to specific MMFF aromaticity related rules that perceive benzene, pyridine, furan, etc... as aromatic molecules while nucleobases and other etherocyclic molecules as not aromatic.
As a last question I would like to ask if you have a(some) reference(s) where I may find specifically MMFF aromaticity model explained? In particular a reference where these MMFF criteria/rules are rationalized/listed in order to properly assign aromaticity according to MMFF aromaticity model without reverse engineering `setMMFFAromaticity` function. Unfortunately, as far as I read, I didn't find these rules listed in Halgren's MMFF94 papers.

Thank you again for your kind attention!
